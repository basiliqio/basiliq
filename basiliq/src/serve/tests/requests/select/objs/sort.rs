use super::*;

crate::run_test_request!(
    sort_local_single,
    Method::GET,
    "/public__peoples?sort=first-name",
    200
);

crate::run_test_request!(
    sort_local_multi,
    Method::GET,
    "/public__peoples?sort=-first-name",
    200
);

crate::run_test_request!(
    sort_remote_single,
    Method::GET,
    "/public__peoples?sort=public__favorite_color.color",
    200
);

crate::run_test_request!(
    sort_remote_single_multi,
    Method::GET,
    "/public__peoples?sort=-public__favorite_color.color,-first-name",
    200
);
