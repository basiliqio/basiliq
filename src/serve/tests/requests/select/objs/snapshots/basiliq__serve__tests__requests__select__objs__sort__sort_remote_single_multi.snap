---
source: basiliq/src/serve/tests/requests/select/objs/sort.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": [
    {
      "type": "public__peoples",
      "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f",
      "attributes": {
        "age": 22,
        "first-name": "Francis",
        "gender": "M",
        "last-name": "Le Roy",
        "twitter": null
      }
    },
    {
      "type": "public__peoples",
      "id": "961e543a-4b22-4d48-a8e5-c1eafada950f",
      "attributes": {
        "age": null,
        "first-name": "AAAAAAAA",
        "gender": null,
        "last-name": "BBBBBBBBB",
        "twitter": null
      }
    },
    {
      "type": "public__peoples",
      "id": "777cc565-c66b-4942-ab44-8fc5f194b804",
      "attributes": {
        "age": 34,
        "first-name": "Somebody",
        "gender": "F",
        "last-name": "Wuhu",
        "twitter": "@randomhandle"
      }
    }
  ]
}
